Documentation/contributing/making-changes.rst

```diff
@@ -0,0 +1,256 @@
+
+    .. code-block:: bash
+
+       $ ./tools/checkpatch.sh -h
+       USAGE: ./tools/checkpatch.sh [options] [list|-]
+
+       Options:
+       -h
+       -c spell check with codespell(install with: pip install codespell
+       -r range check only (used with -p and -g)
+       -p <patch list> (default)
+       -g <commit list>
+       -f <file list>
+       -  read standard input mainly used by git pre-commit hook as below:
+          git diff --cached | ./tools/checkpatch.sh -
+
+   Run it against your files and correct all the errors in the code you added, so that
+   ``tools/checkpatch.sh`` reports no errors. Then commit the result.
+   For example:
+
+    .. code-block:: bash
+
+       $ ./tools/checkpatch.sh -f my-file.c
+       arch/arm/src/sama5/hardware/my-file.c:876:82: warning: Long line found
+       $ # fix errors
+       $ vim my-file.c
+       $ # run again
+       $ ./tools/checkpatch.sh -f my-file.c
+
+   If you have made a lot of changes, you can also use this bash commandline to see the errors for all the changed C
+   files in your branch (assumes you are currently on the branch that has the changed files):
+
+    .. code-block:: bash
+
+       $ git diff --name-only master | egrep "\.c|\.h" | xargs echo | xargs ./tools/checkpatch.sh -f | less
+
+   Note that there are some bugs in the ``nxstyle`` program that ``checkpatch.sh`` uses, so
+   it may report a few errors that are not actually errors. The committers will help you
+   find these. (Or view the
+   `nxstyle Issues <https://github.com/apache/incubator-nuttx/issues?q=is%3Aissue+is%3Aopen+nxstyle>`_.)
+   |br|
+   |br|
+
+#. Commit the fixed files
+
+    .. code-block:: bash
+
+       $ git add my-file.c
+       $ git commit my-file.c
+       $ git push
+
+
+Submitting Your Changes to NuttX
+--------------------------------
+
+  Pull requests let you tell others about changes you've pushed to a branch in a repository on GitHub. Once a pull
+  request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits
+  before your changes are merged into the base branch.
+
+  (from GitHub's `About pull requests <https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests>`_ page)
+
+Before you do a Pull Request, the NuttX team will usually want all the changes you made in your branch "squashed" into
+a single commit, so that when they review your changes, there's a clean view of the history. If there are changes
+after Pull Request review feedback, they can be separate commits. Here's the easiest way I found to do that initial
+squash before submitting the Pull Request:
+
+#. Check out my branch
+
+    .. code-block:: bash
+
+       $ git checkout my-branch
+
+#. Rename it to ``my-branch-old`` to save it, because we're going to create new clean branch to push:
+
+    .. code-block:: bash
+
+       $ git branch -m my-branch-old
```

Comment:
I think interactive rebasing is a better practice to achieve same.

Reply:
I agree

Reply:
@Ouss4 @v01d Will you post a rebase flow in reply to this comment, and I'll update the docs to have this flow as the first option. 

I use the flow that's shown here, and it seems to be equivalent, so I'd like to leave it in as a second option. Do you have objections to that?

Reply:
@Ouss4 @btashton I got the rebase flow out of the other comment block, so you don't need to do anything.